xl_auth/utils.py

```diff
@@ -3,11 +3,31 @@
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-from flask import flash
+from flask import flash, request
+
+try:
+    # noinspection PyCompatibility
+    from urlparse import urlparse, urljoin
+except ImportError:
+    # noinspection PyCompatibility
+    from urllib.parse import urlparse, urljoin
 
 
 def flash_errors(form, category='warning'):
     """Flash all errors for a form."""
     for field, errors in form.errors.items():
         for error in errors:
             flash('{0} - {1}'.format(getattr(form, field).label.text, error), category)
+
+
+def get_redirect_target():
+    """Return safe redirect URL, from `<http://flask.pocoo.org/snippets/62/>`_."""
+    for target in [request.form.get('next_redirect'), request.values.get('next')]:
+        if not target:
+            continue
+        ref_url = urlparse(request.host_url)
+        test_url = urlparse(urljoin(request.host_url, target))
+        if test_url.scheme in ('http', 'https') and ref_url.netloc == test_url.netloc:
+            return target
```

Comment:
What would the implications be if we returned `test_url` here instead? Would it be nice to only have absolute paths when redirecting?

Reply:
@mxtthias I think that would be a bit intrusive; the purpose of the function is only to return the redirect URL— after safety vetting —and not mutating it. The default behavior for Flask is to use (and generate) relative URLs, it think we shouldn't attempt to outsmart it!